distributed/scheduler.py

```diff
@@ -654,6 +657,79 @@ def _to_dict_no_nest(self, *, exclude: Container[str] = ()) -> dict[str, Any]:
             members=True,
         )
 
+    def add_to_processing(self, ts: TaskState) -> None:
+        """Assign a task to this worker for compute."""
+        assert self.scheduler_ref and (scheduler := self.scheduler_ref())
```

Comment:
We do this in a couple of places, I'd factor it out into a dedicated method/property.